service/src/main/kotlin/fi/espoo/evaka/invoicing/service/NewDraftInvoiceGenerator.kt

```diff
@@ -289,19 +305,28 @@ class NewDraftInvoiceGenerator(private val productProvider: InvoiceProductProvid
         price: Int,
         codes: DaycareCodes,
         dailyFeeDivisor: Int,
+        contractDaysPerMonth: Int?,
         attendanceDates: List<LocalDate>,
         feeAlterations: List<Pair<FeeAlteration.Type, Int>>,
         absences: List<AbsenceStub>
     ): List<InvoiceRow> {
-        val periodOperationalDates = attendanceDates.filter { period.includes(it) }
-        if (periodOperationalDates.isEmpty()) return listOf()
+        val periodAttendanceDates = attendanceDates.filter { period.includes(it) }
+        if (periodAttendanceDates.isEmpty()) return listOf()
+
+        // Make sure the number of operational days in a month doesn't exceed `dailyFeeDivisor`.
+        //
+        // Example: A child has a placement to a round-the-clock unit for the first half and to a
+        // normal unit for the second half of the month. The round-the-clock unit has more operational
+        // days, so we have to make sure that we don't invoice more than `dailyFeeDivisor` days.
+        //
+        val attendanceDaysLeft = attendanceDates.take(dailyFeeDivisor).filter { period.includes(it) }.size
```

Comment:
Do we need `periodAttendanceDates` anymore?

Reply:
Not sure what you mean. They are required for computing how many "assumed attendance days" this period has of all the days in the month.

Reply:
I mean why don't we use
```
attendanceDates.take(dailyFeeDivisor).filter { period.includes(it) }
```
in place of 
```
attendanceDates.filter { period.includes(it) }
```

Reply:
Maybe we should discuss this somewhere else.